### ISSUE TITLE
Multi threading in OpenCV/Numpy

### ISSUE THREAD

**Issue by emlynjdavies** [SINTEF/PySilCam#29]
Some has experienced OpenCV being slow due too it spawning too many threads, ref:  http://www.anoopjohnson.com/2010/10/opencv-performance-and-threads/

I did confirm this by doing som small test, running debayering with 6 threads was faster than 16 threads, and uses a lot less processing power (approx 1000% less).

OpenCV was able to debayer around 200 images per second with one thread, so setting cv2.setNumThreads(1) should give some improvements

 

It looks like it is possible to set number of Numpy threads as well: https://stackoverflow.com/questions/30791550/limit-number-of-threads-in-numpy

**Comment by emlynjdavies** [comment]
Disable numpy (really MKL library) threading: https://stackoverflow.com/questions/17053671/python-how-do-you-stop-numpy-from-multithreading


Threads and processes issues and solutions (with logging): https://codewithoutrules.com/2018/09/04/python-multiprocessing/

TLDR – avoid fork() without clearing/resetting the child process (execve()):

“””

from multiprocessing import get_context

 

def your_func():

    with get_context("spawn").Pool() as pool:

        # ... everything else is unchanged

“””

Raymond Nepstad
